workshops/discord_chatbot/README.md

```diff
@@ -0,0 +1,372 @@
+
+Once that reply happens, we will capture that in a variable called `responses`. From here we will log the `queryText` which is the text what the user sent, and return the `fulfillmentText` which is a response to that text got back from Dialogflow.
+
+Whew, that's a lot to digest huh? But stay with me, we're almost on the way to complete this!
+
+<hr />
+
+Now what we need to do is integrate both Dialogflow and discord.js to make a chatbot. For this, you only need to change the function when a message reaches the bot. Instead of sending "Pong!", we will send the text got from DIalogflow.
+
+```js
+client.on('message', (message) => {
+  if (!message.author.bot) {
+    replyMsg(message.content).then((res) => {
+      console.log(res)
+      message.reply(res)
+    })
+  }
+})
+```
+
+So here, we are first checking if the user sending this message is not a bot. If you don't include this condition, then there may be an infinite loop where the bot replies to itself. So keep it for safety.
+
+After the checking, you relay the message sent from the user to DIalogflow. And when the response is received by Dialogflow, you are sending it back to the user.
+
+It's all coming together now eh?
+
+![coming together](https://media.giphy.com/media/KFszZKN9kP8UJk7BlS/giphy.gif)
+
+## To try it out!
+
+Now to finally try out the bot and see if it actually works!
+
+Check your code if all your variables and JSON api files are correct and updated. If all looks good, hit the RUN button, and wait for the `Bot ready!` message to be displayed. If this is displayed, then your discord API is working!
+
+Now, try sending a message in the server.
+
+This is what I received from the bot when I tried to talk with it
+
+![bot texting](https://cloud-cl244cmpv.vercel.app/screenshot14.png)
+
+![done](https://media.giphy.com/media/Q5RiiXiUWWVjvE3rER/giphy.gif)
+
+It's awesome isn't it? You can now talk with your virtual agent you created in this workshop!
+
+## Going forward
+
+You can go back to your dialogflow console and customise your bot even more! You can go back to the code and documentation and see if there is any new features and stuff which can be added or modified.
+
+Here are the codes some demos of how the project can be expanded upon:
+
+- Bot which replies to the user about Hack Club: [code](https://repl.it/@k4u5h4L/discord-chatbot-complete)
+- Bot which sends a scraped meme when a user asks for one: [code](https://repl.it/@k4u5h4L/discord-chatbot-complete-meme)
+- Bot which tells you joke when called: [code](https://repl.it/@k4u5h4L/discord-chatbot-complete)
+
+## Note:
+
+- If I may have disclosed any API keys in this workshop, please don't use the same keys since I have already revoked them and they don't work anymore. You have to visit those sites and get registered to get your hands on these keys.
```

Comment:
I think you should remove this. It's kinda weird, and also I don't think you disclosed any API keys. But also you should make sure you didn't disclose any sensitive info